File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/state/orderstate/OrderContext.java
// OrderContext.java

package com.bigboxcode.designpattern.state.orderstate;

public class OrderContext {
    private OrderState state;
    private OrderState orderPlacedState;
    private OrderState orderInProgressState;
    private OrderState orderDeliveredState;
    private OrderState orderReceivedState;

    public OrderContext() {
        orderPlacedState = new OrderCheckState(this);
        orderInProgressState = new OrderInProgressState(this);
        orderDeliveredState = new OrderDeliverState(this);
        orderReceivedState = new OrderReceiveState(this);

        // Set the placed state as default
        state = orderPlacedState;
    }

    public void setState(OrderState state) {
        this.state = state;
    }

    public OrderState getState() {
        return state;
    }

    public OrderState getOrderPlacedState() {
        return orderPlacedState;
    }

    public OrderState getOrderInProgressState() {
        return orderInProgressState;
    }

    public OrderState getOrderDeliveredState() {
        return orderDeliveredState;
    }

    public OrderState getOrderReceivedState() {
        return orderReceivedState;
    }

    public void runNextProcess() {
        if (state != null) {
            state.process();
        } else {
            System.out.println("Order processing complete");
        }
    }
}

File: design-patterns/php/state/order/OrderContext.php
<?php
// OrderContext.php

namespace BigBoxCode\DesignPattern\State\Order;


class OrderContext {
    private ?OrderState $state;
    private OrderState $orderCheckState;
    private OrderState $orderInProgressState;
    private OrderState $orderDeliverState;
    private OrderState $orderReceiveState;

    public function __construct() {
        $this->orderCheckState = new OrderCheckState($this);
        $this->orderInProgressState = new OrderInProgressState($this);
        $this->orderDeliverState = new OrderDeliverState($this);
        $this->orderReceiveState = new OrderReceiveState($this);

        // Set the placed state as default
        $this->state = $this->orderCheckState;
    }

    public function setState(?OrderState $state): void {
        $this->state = $state;
    }

    public function getState(): OrderState|null {
        return $this->state;
    }

    public function getOrderCheckState(): OrderState {
        return $this->orderCheckState;
    }

    public function getOrderInProgressState(): OrderState {
        return $this->orderInProgressState;
    }

    public function getOrderDeliverState(): OrderState {
        return $this->orderDeliverState;
    }

    public function getOrderReceiveState(): OrderState {
        return $this->orderReceiveState;
    }

    public function runNextProcess(): void {
        if ($this->state != null) {
            $this->state->process();
        } else {
            echo "Order processing complete\n";
        }
    }
}